lib/src/v3/pool.dart

```diff
@@ -0,0 +1,203 @@
+import 'dart:async';
+
+import 'package:pool/pool.dart';
+import 'package:postgres/postgres_v3_experimental.dart';
+
+import 'connection.dart';
+
+class PoolImplementation implements PgPool {
```

Comment:
Note: I think we could take any/most of this package here, I was planning to merge it into `package:postgres`: https://github.com/agilord/postgres_pool/blob/master/lib/postgres_pool.dart

I don't think we need every part of it right now, just putting it out there for reference.